File: one-unity/core/development/common/game-account/Runtime/Scripts/Base/AuthError.cs
namespace TPFive.Game.Account
{
    public class AuthError
    {
        // Also means success
        public static readonly AuthError OK = new AuthError(0, "OK");

        // System error
        public static readonly AuthError FunctionNotSupported = new AuthError(2001, "Function not supported");
        public static readonly AuthError FunctionNotImplemented = new AuthError(2002, "Function not implemented");

        // Data cache error
        public static readonly AuthError SaveDataFailed = new AuthError(2011, "Save data failed");
        public static readonly AuthError ReadDataFailed = new AuthError(2012, "Read data failed");
        public static readonly AuthError DeleteDataFailed = new AuthError(2013, "Delete data failed");

        // State error
        public static readonly AuthError NotSignedIn = new AuthError(2021, "Not signed in");
        public static readonly AuthError SignInAlready = new AuthError(2022, "Already signed in");

        // Auth error
        public static readonly AuthError InvalidParameter = new AuthError(2101, "Invalid parameter");
        public static readonly AuthError InvalidCredentials = new AuthError(2102, "Invalid credentials");
        public static readonly AuthError InvalidAccessToken = new AuthError(2103, "Invalid access token");
        public static readonly AuthError InvalidRefreshToken = new AuthError(2104, "Invalid refresh token");
        public static readonly AuthError AccessTokenExpired = new AuthError(2105, "Access token expired");
        public static readonly AuthError RefreshTokenExpired = new AuthError(2106, "Refresh token expired");
        public static readonly AuthError UserCancelLogin = new AuthError(2111, "User cancel login");
        public static readonly AuthError AuthCodeFailed = new AuthError(2112, "Failed to get authorization code during login");
        public static readonly AuthError AuthTokenFailed = new AuthError(2113, "Failed to exchange access_token with authorization code");

        // User instance error
        public static readonly AuthError UsernameAlreadyExists = new AuthError(2201, "Username already exists");
        public static readonly AuthError EmailAlreadyExists = new AuthError(2202, "Email already exists");
        public static readonly AuthError UsernameMalformed = new AuthError(2203, "Username malformed");
        public static readonly AuthError EmailMalformed = new AuthError(2204, "Email malformed");
        public static readonly AuthError UserNotFound = new AuthError(2205, "User not found");
        public static readonly AuthError RenewAccessTokenFailed = new AuthError(2206, "Renew access token failed");

        // Http error
        public static readonly AuthError Network = new AuthError(2301, "Network error");
        public static readonly AuthError RequestTimeout = new AuthError(2302, "Request timeout");
        public static readonly AuthError RequestFailed = new AuthError(2303, "Request failed");
        public static readonly AuthError EmptyResponse = new AuthError(2304, "Empty response");
        public static readonly AuthError InvalidResponse = new AuthError(2305, "Invalid response");
        public static readonly AuthError MalformedResponse = new AuthError(2306, "Malformed response");

        // Device auth error
        public static readonly AuthError AuthorizationPending = new AuthError(2501, "Authorization pending");
        public static readonly AuthError Slowdown = new AuthError(2502, "Slowdown");
        public static readonly AuthError CannotGetDeviceCode = new AuthError(2503, "Cannot get device code");
        public static readonly AuthError DeviceCodeExpired = new AuthError(2504, "Device code expired");

        // Guest auth error
        public static readonly AuthError GuestNotExist = new AuthError(2601, "Guest not exist");
        public static readonly AuthError CreateGuestFailed = new AuthError(2602, "Create guest failed");

        public static readonly AuthError Unknown = new AuthError(2999, "Unknown error");

        public AuthError(int code, string message)
        {
            Code = code;
            Message = message;
        }

        public int Code { get; }

        public string Message { get; private set; }
    }
}

File: one-mobile/flutter_project/lib/xrauth/constants/auth_error.dart
class AuthError {
  final int code;
  final String message;

  AuthError(this.code, this.message);

  @override
  String toString() {
    return 'AuthError{code: $code, message: $message}';
  }

  // Also means success
  static final AuthError none = AuthError(0, 'None');

  // System error
  static final AuthError functionNotSupported =
      AuthError(2001, 'Function not supported');
  static final AuthError functionNotImplemented =
      AuthError(2002, 'Function not implemented');

  // Data cache error
  static final AuthError saveDataFailed = AuthError(2011, 'Save data failed');
  static final AuthError readDataFailed = AuthError(2012, 'Read data failed');
  static final AuthError deleteDataFailed =
      AuthError(2013, 'Delete data failed');

  // State error
  static final AuthError notSignedIn = AuthError(2021, 'Not signed in');
  static final AuthError signInAlready = AuthError(2022, 'Already signed in');

  // Auth error
  static final AuthError invalidParameter =
      AuthError(2101, 'Invalid parameter');
  static final AuthError invalidCredentials =
      AuthError(2102, 'Invalid credentials');
  static final AuthError invalidAccessToken =
      AuthError(2103, 'Invalid access token');
  static final AuthError invalidRefreshToken =
      AuthError(2104, 'Invalid refresh token');
  static final AuthError accessTokenExpired =
      AuthError(2105, 'Access token expired');
  static final AuthError refreshTokenExpired =
      AuthError(2106, 'Refresh token expired');
  static final AuthError userCancelLogin = AuthError(2111, 'User cancel login');
  static final AuthError authCodeFailed =
      AuthError(2112, 'Failed to get authorization code during login');
  static final AuthError authTokenFailed = AuthError(
      2113, 'Failed to exchange access_token with authorization code');

  // User instance error
  static final AuthError usernameAlreadyExists =
      AuthError(2201, 'Username already exists');
  static final AuthError emailAlreadyExists =
      AuthError(2202, 'Email already exists');
  static final AuthError usernameMalformed =
      AuthError(2203, 'Username malformed');
  static final AuthError emailMalformed = AuthError(2204, 'Email malformed');
  static final AuthError userNotFound = AuthError(2205, 'User not found');
  static final AuthError renewAccessTokenFailed =
      AuthError(2206, 'Renew access token failed');

  // Http error
  static final AuthError network = AuthError(2301, 'Network error');
  static final AuthError requestTimeout = AuthError(2302, 'Request timeout');
  static final AuthError requestFailed = AuthError(2303, 'Request failed');
  static final AuthError emptyResponse = AuthError(2304, 'Empty response');
  static final AuthError invalidResponse = AuthError(2305, 'Invalid response');
  static final AuthError malformedResponse =
      AuthError(2306, 'Malformed response');

  // Device auth error
  static final AuthError authorizationPending =
      AuthError(2501, 'Authorization pending');
  static final AuthError slowdown = AuthError(2502, 'Slowdown');
  static final AuthError cannotGetDeviceCode =
      AuthError(2503, 'Cannot get device code');
  static final AuthError deviceCodeExpired =
      AuthError(2504, 'Device code expired');

  // Guest auth error
  static final AuthError guestNotExist = AuthError(2601, 'Guest not exist');
  static final AuthError createGuestFailed =
      AuthError(2602, 'Create guest failed');

  static final AuthError unknown = AuthError(2999, 'Unknown error');
}
